src/stream/BidirectionalStream.jl

```diff
@@ -132,16 +150,22 @@ function _is_stream_respart_done(req_result::Transaction_Res_All, bidirect_strea
         req_push = true
         loop_break = false
         @debug "answer to req_id: $(req_result.req_id)"
+    else
+        throw(TypeDBClientException("Not known result type"))
     end
 
     return req_push, loop_break
 end
 
 
 function close(stream::BidirectionalStream)
-    close(stream.input_channel)
-    close(stream.output_channel)
-    close(stream.dispatcher)
-    close(stream.resCollector)
+    try
+        close(stream.input_channel)
+        close(stream.output_channel)
+        close(stream.dispatcher)
+        close(stream.resCollector)
```

Comment:
I would like to comment about all code that involves I/O (and so not just these 4 lines that I highlighted above).

In general, system calls (especially those involving I/O) can fail for any reasons. So the best practice is to wrap EVERY call with a try/catch block. Then, you can decide what to do if that call fails. In terms of closing out resources, it is common to just log and ignore then error when that call fails.

To illustrate a potential issue, in the above 4 lines, if the call `close(stream.input_channel)` fails then the other three `close` calls would be skipped, leading to a memory leak.

With Julia, this can be very easy with duck typing. For example, you can define a `safe_close(x)` function that takes any resource `x` and calls `close(x)` in a try-catch block.